### ISSUE TITLE
Inserting a space and then typing a period should backspace the space before typing the period

### ISSUE THREAD

**Issue by DeflateAwning** [dessalines/thumb-key#172]

**Comment by dessalines** [comment]
I'm not sure I follow. Using the clause-ending spacebar multitaps, tapping the spacebar 3 times should do a period thes a space, so it can start the new sentence.

**Comment by DeflateAwning** [comment]
What I meant was, say I type "Some words " (note the space after the words). Now I decide I want to type a period. If I type a period now, the next turns in to "Some words ." (note the space between the words and the period. The functionality of other virtual keyboards is to automatically remove the space, so the typed text becomes "Some words." (which is how you would generally want it to be).

If you actually want to type "Some words ." for some reason, then you have to type "Some words<space><space>." (note the two spaces). 

I'd suggest this should be default behavior, but maybe there should be a setting to disable it if people don't like it

**Comment by skorokithakis** [comment]
Agreed, that sounds useful.

**Comment by dessalines** [comment]
This seems not-explicit behavior. There are many cases where you might want a period after a space, such as when you use parentheses like `some text (some other text) .` 

If someone really wants this, I'd accept a PR to add that setting, but it seems easier to me to just tap backspace.

**Comment by DeflateAwning** [comment]
Why would someone want to type "some text (some other text) ." I've wanted to type NOT that (i.e., no space) 100x more than I've wanted to type that. 

You can still type it your way; you just have to put an extra space. PR incoming; can you please re-open the issue?

**Comment by skorokithakis** [comment]
I agree, I've never seen a space before a period. The decision that it's more common than the alternative indicates to me that the project needs more focus on better UX decisions.

**Comment by xxxxme** [comment]
If this is added, please make it an option rather than forced.  On the previous keyboard I used, it was very annoying that it autocorrected "sometext : )" to "sometext: )" because of removing pre-punctuation spaces. 
Even if this only applies to periods, I'd actually much prefer to see Thumb-Key stay as simple as possible with as little auto/"error" correction as possible, trusting the user to know what they're doing.

**Comment by DeflateAwning** [comment]
It would only be for periods and commas. Please give an example of a time you've ever intended to insert a space between a letter and a period/comma.

**Comment by xxxxme** [comment]
If it's only for periods and commas, I would not object ,.';.;'.,

**Comment by DeflateAwning** [comment]
Would still like this feature!

**Comment by BlueDrink9** [comment]
Chiming in with example usecase for leaving the space, since others seem confused why this should be optional.

When discussing programming or talking about shell commands (eg in github comments!) I often need to say things like 

```
Select x
, y
, z
From table
```

Or run `ls -l .`

**Comment by dessalines** [comment]
Programming is so different from regular language typing, that it seems like turning off multitaps (and probably autocapitalize) is the best option.

The alternative is to add all kinds of settings for multitap exceptions and intracacies.

**Comment by dessalines** [comment]
Its been a year and no one wanted to work on this. If anyone's willing, I'll re-open and assign it to them.
